core/sql/optimizer/NAColumn.cpp

```diff
@@ -838,3 +838,27 @@ Int32 NAColumnArray::getColumnPosition(NAColumn& nc) const
   return -1;
 }
 
+NAString NAColumnArray::getColumnNamesAsString(char separator) const
+{
+   return getColumnNamesAsString(separator, entries());
+}
+
+NAString NAColumnArray::getColumnNamesAsString(char separator, UInt32 ct) const
+{
+  NAString nmList;
+
+  if ( ct == 0 )
+    return NAString();
+
+  if ( ct > entries() )
+    ct = entries();
+
+  for (CollIndex i = 0; i < ct-1; i++)
+  {
+     nmList += at(i)->getColName();
```

Comment:
The rework has been submitted.

On Thu, Oct 29, 2015 at 11:36 AM, Qifan Chen <qifan.chen@esgyn.com> wrote:

> Good point. Will wrap getColName() with ToAnsiIdentifier().
>
> See
>
> const NAType *Assign::synthesizeType()
> {
>   ValueId targetId, sourceId;
>   const NAType * result = doSynthesizeType(targetId, sourceId);
>   if (result == NULL)
>   {
>     emitDyadicTypeSQLnameMsg(-4039,
>                              targetId.getType(),
>                              sourceId.getType(),
>
>  ToAnsiIdentifier(targetId.getNAColumn()->getColName()));
>     return NULL;
>   }
>   //
>   // Return the result.
>   //
>   return result;
> }
>
>
> On Thu, Oct 29, 2015 at 10:50 AM, DaveBirdsall <notifications@github.com>
> wrote:
>
>> In core/sql/optimizer/NAColumn.cpp
>> <https://github.com/apache/incubator-trafodion/pull/152#discussion_r43403344>
>> :
>>
>> > +   return getColumnNamesAsString(separator, entries());
>> > +}
>> > +
>> > +NAString NAColumnArray::getColumnNamesAsString(char separator, UInt32 ct) const
>> > +{
>> > +  NAString nmList;
>> > +
>> > +  if ( ct == 0 )
>> > +    return NAString();
>> > +
>> > +  if ( ct > entries() )
>> > +    ct = entries();
>> > +
>> > +  for (CollIndex i = 0; i < ct-1; i++)
>> > +  {
>> > +     nmList += at(i)->getColName();
>>
>> Are there any oddities to consider when using delimited identifiers?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/apache/incubator-trafodion/pull/152/files#r43403344>.
>>
>
>
>
> --
> Regards, --Qifan
>
>


-- 
Regards, --Qifan